### PR TITLE
Added GCS sensor in airflow-gfw

### PR DESCRIPTION
**Pull request by smpiano**
Related with> https://github.com/GlobalFishingWatch/GFW-Tasks/issues/1164

### CODE REVIEW

File: airflow_ext/gfw/models.py [hunk 1]
```diff
@@ -90,5 +92,51 @@ def source_table_sensors(self, dag):
...
+                    project=self.config['project_id'],
+                    dataset=self.config['source_dataset'],
+                    table=path,
+                    date=self.source_sensor_date_nodash()
+                )
+        [gcs, tables]
+
+    def source_sensors(self, dag, date=None):
+        return [
+            self.gcs_sensor(dag=dag, **parts) if ('bucket' in parts and 'prefix' in parts)
```
Comment (enriquetuya) [inline airflow_ext/gfw/models.py]: I don't like the idea of mixing the BQ and GCS sensors, I think you are adding code complexity for a case that won't happen any time soon, having both BQ and GCS sources for a given pipeline. I would add the new logic in `source_gcs_sensors`.

File: airflow_ext/gfw/models.py [hunk 2]
```diff
@@ -90,5 +92,51 @@ def source_table_sensors(self, dag):
...
+            bucket=bucket,
+            prefix='{}/{}'.format(prefix, date),
+            mode='reschedule',      # the sensor task frees the worker slot when the criteria is not yet met
+                                    # and it's rescheduled at a later time.
+            poke_interval=10 * 60,  # check every 10 minutes.
+            timeout=60 * 60 * 24    # timeout of 24 hours.
+        )
+
+    def source_path(self, date=None):
+        gcs_paths = self.config.get('source_paths') or self.config.get('source_path')
```
Comment (enriquetuya) [inline airflow_ext/gfw/models.py]: As my comment below, I don't like mixing table paths and gcs path. I would keep `source_path` as it is and add `source_gcs_path`

**Review by enriquetuya — CHANGES_REQUESTED** [review]
@smpiano check my comments below. I don't like mixing BQ things with GCS things. I know where you are going but I don't think is the best to have `sources` with a mix of tables and gcs.. we haven't had that scenario yet. 

Also, add more documentation to the methods.

File: airflow_ext/gfw/models.py (hunk 2 above)
Comment (smpiano) [reply airflow_ext/gfw/models.py]: @enriquetuya I will work on the changes to have only `source_gcs_path`

File: airflow_ext/gfw/models.py (hunk 1 above)
Comment (smpiano) [reply airflow_ext/gfw/models.py]: Understood.

File: airflow_ext/gfw/models.py (hunk 2 above)
Comment (rrequero) [reply airflow_ext/gfw/models.py]: I think the same. In my opinion, it's better to separate in 2 fields.

File: airflow_ext/gfw/models.py (hunk 1 above)
Comment (rrequero) [reply airflow_ext/gfw/models.py]: Yes, better separate the code in 2 functions. And the Dag that extends DagFactory invoke the method that it needs

**Pull request merged**